lms/djangoapps/certificates/management/commands/cert_whitelist.py

```diff
@@ -12,6 +12,18 @@
 from django.contrib.auth.models import User
 
 
+def get_user_from_identifier(identifier):
+    """
+     This function takes the string identifier and fetch relevant user object from database
+    """
+    identifier = identifier.strip()
+    if '@' in identifier:
```

Comment:
There's also django's built in email validator: http://stackoverflow.com/questions/3217682/checking-validity-of-email-in-django-python.

If we ever allow non-alphanumeric characters in usernames, what's to stop someone from setting their username to someone else's email address?